src/Mnemonic.cpp

```diff
@@ -28,16 +28,15 @@ inline const char* const* mnemonicWordlist() { return wordlist; }
 bool Mnemonic::isValidWord(const std::string& word) {
     const char* wordC = word.c_str();
     const auto len = word.length();
+    // Although this operation is not security-critical, we aim for constant-time operation here as well
+    // (i.e., no early exit on match)
+    auto found = false;
     for (const char* const* w = mnemonicWordlist(); *w != nullptr; ++w) {
-        if (strlen(*w) != len) {
-            continue;
-        }
-        if (strncmp(*w, wordC, len) == 0) {
-            return true;
+        if (strlen(*w) == len && strncmp(*w, wordC, len) == 0) {
```

Comment:
Is this really constant? C++ uses short-circuit evaluation tho, `strncmp` won't be called if `strlen(*w) == len` is false

Reply:
Yeah, even though this fix makes analysis harder, we can still measure comp time for each possible word and then use this info for attack

Reply:
True points. Correct would be to pad every word and the input to a fixed size (e.g. 100 chars), and work with those.
However, let's keep it like this.